### PR TITLE
style: update type annotations

### PR DESCRIPTION
**Pull request by atsushi421**
## Description
This PR updates the type hint description.
The following files contain a few modifications other than the type hint descriptions.
- package.xml
- requirements.txt
- test_type_check_decorator.py

The reason for the modification of the above file is that the version of [pydantic installed by rosdep install](https://ubuntu.pkgs.org/22.04/ubuntu-universe-arm64/python3-pydantic_1.8.2-1_all.deb.html) does not support the new type hint description method, so it was changed to pip install.

## Related links
- Jira ticket: https://tier4.atlassian.net/browse/RT2-713
- https://peps.python.org/pep-0585/

## Notes for reviewers
There is no degradation using CARET_report.
image

## Pre-review checklist for the PR author

- [x] I've confirmed the [contribution guidelines](https://github.com/tier4/caret/blob/main/.github/CONTRIBUTING.md).
- [x] The PR is ready for review.

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR has been properly tested.
- [x] The PR has been reviewed.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [x] (Optional) The PR has been properly tested with CARET_report verification.
- [x] There are no open discussions or they are tracked via tickets.
- [x] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

### CODE REVIEW

File: src/test/common/test_type_check_decorator.py [hunk 1]
```diff
@@ -12,14 +12,20 @@
...
 
 import pytest
 
 
+class DummyCustom:
+
+    def __init__(self) -> None:
+        pass
+
+
```
Comment (tamegaictc2) [inline src/test/common/test_type_check_decorator.py:28]: What is the reason for this change?

File: src/package.xml [hunk 2]
```diff
@@ -11,7 +11,6 @@
   <exec_depend>tracetools_analysis</exec_depend>
   <exec_depend>python3-tqdm</exec_depend>
   <exec_depend>python3-yaml</exec_depend>
-  <exec_depend>python3-pydantic</exec_depend>
```
Comment (tamegaictc2) [inline src/package.xml:14]: What is the reason for this change?

File: src/caret_analyze/infra/lttng/lttng.py [hunk 3]
```diff
@@ -15,12 +15,13 @@
 from __future__ import annotations
 
 from abc import ABCMeta, abstractmethod, abstractproperty
+from collections.abc import Iterable, Iterator, Sequence, Sized
```
[the remark this answers is not in the document]
Comment (atsushi421) [reply src/caret_analyze/infra/lttng/lttng.py:18]: The change to collections.abc seems correct as Sized is inherited to use the `__len__` method.
https://docs.python.org/ja/3/library/collections.abc.html#collections.abc.Sized

**Review by tamegaictc2 — DISMISSED** [review]
I have reviewed all the changes.

**Review by tamegaictc2 — APPROVED** [review]
I have reviewed all the modifications based on the following aspects:
- Whether the processing remains unchanged between before and after the modification.
- Whether there are any missed changes. 
- Whether there are any unnecessary modifications.

**Comment by atsushi421** [comment]
@nabetetsu CC: @takeshi-iwanari 
Now that EmbIV's internal review is complete, can you review this PR?

**Review by nabetetsu — DISMISSED** [review]
I have confirmed there are not any unnecessary changes.
I have also confirmed all files that there are no ommisions.

**Comment by atsushi421** [comment]
@nabetetsu 
I'm sorry, I have resolved the competition. Could you please approve it again?

**Review by nabetetsu — APPROVED** [review]
LGTM :+1:
